src/main/java/se/sundsvall/snailmail/service/BatchScheduler.java

```diff
@@ -0,0 +1,53 @@
+package se.sundsvall.snailmail.service;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import se.sundsvall.dept44.requestid.RequestId;
+import se.sundsvall.snailmail.integration.db.BatchRepository;
+
+@Component
+public class BatchScheduler {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(BatchScheduler.class);
+
+	private final Duration outdatedAfter;
+	private final BatchRepository batchRepository;
+	private final SnailMailService snailMailService;
+
+	public BatchScheduler(@Value("${batch.dangling.outdated-after}") final Duration outdatedAfter, BatchRepository batchRepository, SnailMailService snailMailService) {
```

Comment:
Use `@ConfigurationProperties` and `@Validated` in conjunction with appropriate `jakarta.validation.*` constraints for a basic sanity/null check on the `batch.dangling.*` properties?